[lldb-dap] Fix segfault in JSONUtils.cpp when GetUUIDString() returns nullptr

aahrun opened the pull request:
When creating a stack frame in JSONUtils.cpp CreateStackFrame() the code constructs a std::string from module.GetUUIDString(), which can return nullptr in some cases (as documented in the implementation of SBModule::GetUUIDString()). This causes a segmentation fault when passed to the std::string constructor.

This fix adds a null check before constructing the UUID string, falling back to an empty string if nullptr is returned. The existing empty check ensures the moduleId field is omitted from the JSON when no UUID exists.

aahrun committed:
[lldb-dap] Fix segfault in JSONUtils.cpp when GetUUIDString() returns nullptr

When creating a stack frame in JSONUtils.cpp CreateStackFrame() the code
constructs a std::string from module.GetUUIDString(), which can return
nullptr in some cases (as documented in the implementation of
SBModule::GetUUIDString()). This causes a segmentation fault when passed to
the std::string constructor.

This fix adds a null check before constructing the UUID string, falling back
to an empty string if nullptr is returned. The existing empty check ensures
the moduleId field is omitted from the JSON when no UUID exists.

DavidSpickett commented on lldb/tools/lldb-dap/JSONUtils.cpp:
```diff
@@ -554,7 +554,8 @@ llvm::json::Value CreateStackFrame(DAP &dap, lldb::SBFrame &frame,
-    std::string uuid = module.GetUUIDString();
+    const char *uuid_cstr = module.GetUUIDString();
+    std::string uuid = uuid_cstr ? uuid_cstr : "";
     if (!uuid.empty())
```
You could instead do:
```
if (uuid_cstr) {
 ..., std::string(uuid_cstr)
```

DavidSpickett reviewed (commented):
Looks fine, only thing is whether it's easy to add a test for this, probably too much effort.

Nominate @da-viper as an lldb-dap expert to decide that.

aahrun committed:
Update lldb/tools/lldb-dap/JSONUtils.cpp

Co-authored-by: Ebuka Ezike <yerimyah1@gmail.com>

aahrun committed:
Applied feedback from PR
Better handling of module_id in CompileUnitsRequestHandler.cpp
as per recommendation.

alycm commented on lldb/tools/lldb-dap/Handler/CompileUnitsRequestHandler.cpp:
```diff
@@ -10,6 +10,7 @@
 #include "EventHelper.h"
 #include "JSONUtils.h"
 #include "RequestHandler.h"
+#include "llvm/ADT/StringRef.h"
```
I suspect this line is not required as `GetString` was already used and returns a `StringRef`?

da-viper reviewed (approved):
LGTM! 
Modulo the `#include`

aahrun committed:
Removed unnecessary include
As recommended in PR
